platform/platform_tests_support/test_socket.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "platform/socket.hpp"
+
+#include "std/vector.hpp"
+
+namespace platform
+{
+namespace tests_support
+{
+class TestSocket : public Socket
+{
+public:
+  // Simulates server reading.
+  // Waits for some data or timeout.
+  // Returns size of read data.
+  virtual size_t ReadServer(vector<uint8_t> & destination, uint64_t timeoutMs = 1000) = 0;
+};
+
+unique_ptr<TestSocket> createTestSocket();
```

Comment:
Do we need this function at all? It's name is obvious, it's implementation is trivial?